### PR TITLE
PTCD-830 T Level Detail operation

### PR DESCRIPTION
**Pull request by gunndabad**

### CODE REVIEW

File: src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        public async Task<OneOf<NotFound, TLevelDetailViewModel>> Handle(Query request, CancellationToken cancellationToken)
+        {
+            var tLevel = await _sqlQueryDispatcher.ExecuteQuery(new GetTLevel() { TLevelId = request.TLevelId });
+
+            if (tLevel == null)
+            {
+                return new NotFound();
+            }
+
+            var getProvider = _cosmosDbQueryDispatcher.ExecuteQuery(
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs:41]: Why do we need to use the Cosmos Provider collection? Is everything not in SQL?

File: src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                StudyMode = CourseStudyMode.FullTime,
+                TLevelId = tLevel.TLevelId,
+                Website = tLevel.Website,
+                WhatYouCanDoNext = tLevel.WhatYouCanDoNext,
+                WhatYoullLearn = tLevel.WhatYoullLearn,
+                WhoFor = tLevel.WhoFor
+            };
+        }
+
+        private static string EnsureHttpPrefixed(string url) => !string.IsNullOrEmpty(url)
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs:124]: This is duplicated from CourseDetail... what to do with it I wonder? Extension method on string, scoped to FAC? Static within FAC? (not named Helper, 😄) I'm not sure.

File: src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelLocationViewModel.cs [hunk 3]
```diff
@@ -1,7 +1,10 @@
-﻿namespace Dfc.CourseDirectory.FindACourseApi.Features.TLevelDetail
+﻿using System;
+
+namespace Dfc.CourseDirectory.FindACourseApi.Features.TLevelDetail
 {
     public class TLevelLocationViewModel
     {
+        public Guid TLevelLocationId { get; set; }
```
Comment (jameslawesynergy) [inline src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelLocationViewModel.cs:7]: I can't see any updates to the `SELECT` statements in any SqlQueryHandlers to populate this value?

File: src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs (hunk 2 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs:124]: I think this used to be in an `HttpUtil` class. I'm open to naming suggestions but I'm not crazy about making it an extension method on `string`. Could make it an extension method on `Flurl.Url` and use that here (we use Flurl elsewhere already)?

File: src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs (hunk 1 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs:41]: Mainly for symmetry with the CourseRunDetail one. The current SQL queries don't retrieve `Contacts`, which is required here. In this interest of getting this out of the door quickly I did the simplest thing.

File: src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelLocationViewModel.cs (hunk 3 above)
Comment (gunndabad) [reply src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelLocationViewModel.cs:7]: As far as I can see this property is already populated by the query handlers.

Comment (jameslawesynergy) [reply src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelLocationViewModel.cs:7]: Ignore me, I misunderstood that class.

File: src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs (hunk 2 above)
Comment (jameslawesynergy) [reply src/Dfc.CourseDirectory.FindACourseApi/Features/TLevelDetail/TLevelDetail.cs:124]: Your call, happy to come back to it later if the solution isn't apparent right now.

**Review by jameslawesynergy — APPROVED** [review]

**Review by soharadesign — APPROVED** [review]

**Pull request merged**